### PR TITLE
port `gatk4/createsequencedictionary` to nf-test

### PR DESCRIPTION
**Pull request by famosab**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

Relates to #3937

### CODE REVIEW

File: modules/nf-core/gatk4/createsequencedictionary/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                    ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.versions).match("versions") },
+                { assert snapshot(file(process.out.dict.get(0).get(1)).name).match() }
```
Comment (maxulysse) [inline modules/nf-core/gatk4/createsequencedictionary/tests/main.nf.test:28]: while not snapshotting the file?

Comment (famosab) [reply modules/nf-core/gatk4/createsequencedictionary/tests/main.nf.test]: I changed it to read the lines. The md5sum changes between docker and conda for me.

File: modules/nf-core/gatk4/createsequencedictionary/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                    ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.versions).match("versions") },
+                { assert snapshot(file(process.out.dict.get(0).get(1)).readLines()).match() }
```
Comment (maxulysse) [inline modules/nf-core/gatk4/createsequencedictionary/tests/main.nf.test:28]: Why not just snapshot process.out.dict?

Comment (famosab) [reply modules/nf-core/gatk4/createsequencedictionary/tests/main.nf.test:28]: Like this:
`{ assert snapshot(file(process.out.dict)).match() }`?

File: modules/nf-core/gatk4/createsequencedictionary/tests/main.nf.test (hunk 1 above)
Comment (maxulysse) [reply modules/nf-core/gatk4/createsequencedictionary/tests/main.nf.test]: I see.
The old pytest was using the md5sum which is why I was suggesting this approach

Comment (famosab) [reply modules/nf-core/gatk4/createsequencedictionary/tests/main.nf.test]: I tried it again, maybe the CI works? I just ran into weird stuff on my computer

File: modules/nf-core/gatk4/createsequencedictionary/tests/main.nf.test (hunk 2 above)
Comment (famosab) [reply modules/nf-core/gatk4/createsequencedictionary/tests/main.nf.test]: seems to work

**Comment by famosab** [comment]
For some reason now more nf-tests are triggered like summarizedexperiment...

File: modules/nf-core/gatk4/createsequencedictionary/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                    ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.versions).match("versions") },
+                { assert snapshot(process.out.dict).match() }
```
Comment (maxulysse) [inline modules/nf-core/gatk4/createsequencedictionary/tests/main.nf.test:28]:
```suggestion
                { assert snapshot(process.out).match() }
```
This should be enough then

**Comment by maxulysse** [comment]
> For some reason now more nf-tests are triggered like summarizedexperiment...

There must have been a merge at some point, but I see you fixed that with a master merge.
Well done

**Pull request merged**